ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java

```diff
@@ -135,69 +139,120 @@
     }
   }
 
-  private LineReader lineReader;
   private final ObjectMapper objectMapper;
   private final Map<String, CliSpecificCommand> cliSpecificCommands;
-
+  private final KsqlTerminal terminal;
+  private final RowCaptor rowCaptor;
   private OutputFormat outputFormat;
 
-  public Console(final OutputFormat outputFormat, final KsqlRestClient restClient) {
-    Objects.requireNonNull(
-        outputFormat,
-        "Must provide the terminal with a beginning output format"
-    );
-    Objects.requireNonNull(restClient, "Must provide the terminal with a REST client");
+  public interface RowCaptor {
+    void addRow(GenericRow row);
 
-    this.outputFormat = outputFormat;
+    void addRows(List<List<String>> fields);
+  }
 
-    this.cliSpecificCommands = Maps.newLinkedHashMap();
+  public static Console build(final OutputFormat outputFormat, final KsqlRestClient restClient) {
+    final AtomicReference<Console> console = new AtomicReference<>();
+    final Function<String, Boolean> isCliCommand = line -> {
+      final Console theConsole = console.get();
```

Comment:
This will always be null

Reply:
Hummm... how did I miss that?!?!?  Good spot :D